src/components/screens/consul/Proposals/NewProposal.js

```diff
@@ -361,6 +334,36 @@ export const NewProposal = ({ navigation, data, query }) => {
   );
 };
 
+const ImageSelector = ({ control, imageUri, item, onChange, selectImage, value }) => {
+  const { buttonTitle, infoText, label, name } = item;
+
+  useEffect(() => {
+    if (!value) {
+      onChange(imageUri);
+    }
+  }, [imageUri]);
```

Comment:
i do not understand this part when reading.
why is the dependency `imageUri` but the logic is also depending on the value of `value`. why is there a check for `!value`? what if `value` exists? what is the effect used for?

Reply:
According to `imageUri`, I need to run the `onChange` function again, because after I click the button, I do not receive `imageUri` data. After selecting the photo, I can access this data.

The reason I check `!value` is that if I want to edit a proposal with a photo, it automatically sends the value `undefined` to the `onChange` function and the user cannot see the preview of the picture. This check is just a check for editing the proposal.

Reply:
alright thanks, can you please write something down as comment in the code for our future selfs

Reply:
4b6da24309c37c24328a4e90c8d96a9212c89326